File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.py
from collections import Counter
def solution(k, tangerine):
    count = Counter(tangerine)
    sort = sorted(count.items(), key=lambda x: x[1], reverse=True)
    
    answer = 0
    # k = 6, b = 3
    for a, b in sort:
        if b<=k:
            k-=b
            answer+=1
        elif k==0:
            break
        else:
            answer+=1
            break
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
  var count = 0;

  const sizeObject = tangerine.reduce((acc, cur) => {
    acc[cur] = (acc[cur] || 0) + 1;
    return acc;
  }, {});

  let sizeArr = Object.values(sizeObject).sort((a, b) => b - a);

  for (i = 0; i < sizeArr.length; i++) {
    k = k - sizeArr[i];
    count++;
    if (k <= 0) break;
  }

  return count;
}